Improve Saved Prompt Usage

shubhiscoding opened the pull request:
solving #164 
Use all of your saved prompts by typing / and anything you type after / will be used for searching out of your saved prompts
accessible on home screen and also in a conversation.
prompts are sorted based on your favorite and last used prompt.
![Screenshot from 2025-01-28 20-05-49](https://github.com/user-attachments/assets/33c4d1a5-bffe-49bd-b75b-99b8bb65e45d)
Added a character limit on edit prompt option.
![Screenshot from 2025-01-28 18-31-01](https://github.com/user-attachments/assets/2de30a2b-06f2-4cb7-8933-ec9cd805fbd4)

briang123 commented:
@shubhiscoding Thanks for addressing these concerns. Not sure if you did, but wondering if you're able to make the modal window a bit larger so we're not working with a small window for the prompt. With larger amount of text, I'm constantly having to expand the text input to a larger size, but the width doesn't change either. Could be a quality of life change.

shubhiscoding commented:
> @shubhiscoding Thanks for addressing these concerns. Not sure if you did, but wondering if you're able to make the modal window a bit larger so we're not working with a small window for the prompt. With larger amount of text, I'm constantly having to expand the text input to a larger size, but the width doesn't change either. Could be a quality of life change.

You want me to increase Edit prompt modal size right?

briang123 commented:
> > @shubhiscoding Thanks for addressing these concerns. Not sure if you did, but wondering if you're able to make the modal window a bit larger so we're not working with a small window for the prompt. With larger amount of text, I'm constantly having to expand the text input to a larger size, but the width doesn't change either. Could be a quality of life change.
> 
> You want me to increase Edit prompt modal size right?

Yeah, that would be helpful.

If you're able to increase the size of the modal, as well, that would be nice. Maybe a size that is more proportionate to the size of the window. I feel like we just need more ample room to work in that box.

shubhiscoding commented:
Just pushed the changes! This much size seems fine to me isn't it? what do you think?
![image](https://github.com/user-attachments/assets/3d0d822f-2cf5-4f78-a2bd-57c03f07f63c)

briang123 commented:
> Just pushed the changes! This much size seems fine to me isn't it? what do you think? ![image](https://private-user-images.githubusercontent.com/95687419/407394464-3d0d822f-2cf5-4f78-a2bd-57c03f07f63c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgwODEwNzEsIm5iZiI6MTczODA4MDc3MSwicGF0aCI6Ii85NTY4NzQxOS80MDczOTQ0NjQtM2QwZDgyMmYtMmNmNS00Zjc4LWEyYmQtNTdjMDNmMDdmNjNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI4VDE2MTI1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU1OTQ2NWNhMzJiMGQ0MGY0NmRhZDFhOGM1OGRkZWUwMjdmZDU4ZWFhODBhZGFiYTY2ZGFjZWU5NTQ4OTU2NWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XXwRJwyuRWz5n5czAfeNKuz0UAQ8hYux7E9jG85TOJM)

It does look much better to work with. How does it appear on mobile in a responsive view?

shubhiscoding commented:
> Have you tested these changes on a responsive mobile web view?

I just did it works fine! but the saved prompts page doesn't look good on the mobile view, it needs bit of padding, I'll add it!

shubhiscoding commented:
redesigned the saved prompt page for better mobile view:

Old view:
![Screenshot from 2025-01-28 21-57-10](https://github.com/user-attachments/assets/ef2b321e-0f57-4cbc-a03b-89d6180d399f)

New view:
![image](https://github.com/user-attachments/assets/6915c00a-8f1d-479b-8370-fa19b4a826af) ![image](https://github.com/user-attachments/assets/8a32430d-c0b4-45b6-8771-b3f44a526cc9)

briang123 commented:
is it possible to add prompts mid-conversation? Do your updates address that?

briang123 commented on src/app/(user)/saved-prompts/components/edit-prompt-dialog.tsx:
```diff
@@ -79,9 +79,15 @@ export const EditPromptDialog = ({
           value={content}
           rows={4}
           placeholder="Prompt"
+          maxLength={2000}
```
I would suggest using the constant below (maybe it needs to be moved to a more shareable location) that is used by the conversation input instead of hard-coding this value. If it ever changes then it will be covered here.

https://github.com/NeurProjects/neur-app/blob/532393a7d270a681c1d1516e337777320c027eed/src/app/(user)/home/conversation-input.tsx#L16

shubhiscoding replied:
exported this MAX_CHARS and used it in edit-prompt

briang123 commented on src/app/(user)/saved-prompts/components/edit-prompt-dialog.tsx:
```diff
@@ -79,9 +79,15 @@ export const EditPromptDialog = ({
+          `}
         />
+        <div className="text-sm text-gray-500">
+          {content.length}/2000 chars
```
Again, use the constant in place of 2000

shubhiscoding replied:
imported MAX_CHARS from conversation input and used it in edit-prompt

shubhiscoding commented:
> is it possible to add prompts mid-conversation? Do your updates address that? Meaning, once I start a conversation, in any subsequent inputs, if I enter the "/", allow me to use a prompt? This way i can create follow up prompts throughout my conversation instead of only at the start.

I believe this was already a feature, even in a mid conversation you can type / and access the prompts.
![image](https://github.com/user-attachments/assets/de14ac51-1e9b-4aca-b60e-8d3aac557f35)

briang123 commented:
> > is it possible to add prompts mid-conversation? Do your updates address that? Meaning, once I start a conversation, in any subsequent inputs, if I enter the "/", allow me to use a prompt? This way i can create follow up prompts throughout my conversation instead of only at the start.
> 
> I believe this was already a feature, even in a mid conversation you can type / and access the prompts. ![image](https://private-user-images.githubusercontent.com/95687419/407607556-de14ac51-1e9b-4aca-b60e-8d3aac557f35.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgxMjc3OTgsIm5iZiI6MTczODEyNzQ5OCwicGF0aCI6Ii85NTY4NzQxOS80MDc2MDc1NTYtZGUxNGFjNTEtMWU5Yi00YWNhLWI2MGUtOGQzYWFjNTU3ZjM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI5VDA1MTEzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4MDgxNDg5OWM5YzA4ZmM4OWE1YTMwNGYzOGY3ZTM4ZjViOGRmMjczMzRiZTg2MmQ1YjdhNjc4NDZlYjBmMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WyuT6SorlQ7veAkgbVI42ExZYFsSNLKHaLshQLrQUGs)

Sweet 🤦 , all we needed to do was add a "/" to prompt the saved prompts? I had no clue! Would it make sense to add a little tooltip near the input or a label with message "Press / to trigger saved prompts".

shubhiscoding commented:
> > > is it possible to add prompts mid-conversation? Do your updates address that? Meaning, once I start a conversation, in any subsequent inputs, if I enter the "/", allow me to use a prompt? This way i can create follow up prompts throughout my conversation instead of only at the start.
> > 
> > 
> > I believe this was already a feature, even in a mid conversation you can type / and access the prompts. ![image](https://private-user-images.githubusercontent.com/95687419/407607556-de14ac51-1e9b-4aca-b60e-8d3aac557f35.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgxMjc3OTgsIm5iZiI6MTczODEyNzQ5OCwicGF0aCI6Ii85NTY4NzQxOS80MDc2MDc1NTYtZGUxNGFjNTEtMWU5Yi00YWNhLWI2MGUtOGQzYWFjNTU3ZjM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI5VDA1MTEzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4MDgxNDg5OWM5YzA4ZmM4OWE1YTMwNGYzOGY3ZTM4ZjViOGRmMjczMzRiZTg2MmQ1YjdhNjc4NDZlYjBmMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WyuT6SorlQ7veAkgbVI42ExZYFsSNLKHaLshQLrQUGs)
> 
> Sweet 🤦 , all we needed to do was add a "/" to prompt the saved prompts? I had no clue! Would it make sense to add a little tooltip near the input or a label with message "Press / to trigger saved prompts".

Yeah sure I'll add it!

shubhiscoding commented:
Home screen:
![image](https://github.com/user-attachments/assets/57a2b7be-3a5d-41b3-a9d6-f45a9d1e78de) 
Chat screen:
![Screenshot from 2025-01-29 12-09-26](https://github.com/user-attachments/assets/71c17838-7b7c-4d8a-8e99-44738467143d)
